test/fetch.test.js

```diff
@@ -1,14 +1,36 @@
 describe('fetch', () => {
-  let f, fetch
+  let f, fetch, originalOpen, originalFetch
 
   beforeEach(() => {
-    f = jest.fn()
-    jest.doMock('isomorphic-unfetch', () => f)
-    fetch = require('../src/fetch').default
+    jest.isolateModules(() => {
+      f = jest.fn()
+      jest.doMock('isomorphic-unfetch', () => f)
+      originalOpen = jest.spyOn(XMLHttpRequest.prototype, 'open')
+      originalFetch = jest.spyOn(window, 'fetch').mockImplementation()
+      fetch = require('../src/fetch').default
+      window.__NEXT_DATA__ = {
+        buildId: 'development',
+      }
+    })
+  })
+
+  afterEach(() => {
+    delete window.__NEXT_DATA__
   })
 
   it('should export isomorphic-unfetch for backwards compatibility', () => {
     fetch('/foo')
     expect(f).toHaveBeenCalledWith('/foo')
   })
+
+  it('should patch window.fetch', () => {
+    window.fetch('/foo')
```

Comment:
I would add a test for the fetch with request object param as well

Reply:
done